src/common/chains.ts

```diff
@@ -9,3 +11,383 @@ export enum SupportedChainId {
   BASE = 8453,
   SEPOLIA = 11155111,
 }
+
+/**
+ * The chain id of the chain.
+ *
+ * TODO: Should we generalize it even more to allow non-EVM chains? We should probably revisit also the chain interface, and some other types.
+ */
+export type ChainId = number
+
+export interface WebUrl {
+  url: string
+  name: string
+}
+
+export type ChainRpcUrls = {
+  http: readonly string[]
+  webSocket?: readonly string[] | undefined
+}
+
+export type ChainContract = {
+  address: string
+  blockCreated?: number | undefined
+}
+
+export type ChainContracts = {
+  multicall3?: ChainContract
+  ensRegistry?: ChainContract
+  ensUniversalResolver?: ChainContract
+  universalSignatureVerifier?: ChainContract
+}
+
+/**
+ * A chain on the network.
+ *
+ * Probably we could use the viem chain definition, I think multicall3, and ensRegistry and the types defined there can be handy. But for now we are using a simplified version.
+ *
+ * For a list of chains, see: https://github.com/wevm/viem/tree/main/src/chains/definitions
+ */
+export interface ChainInfo {
+  /**
+   * The chain id.
+   */
+  readonly id: ChainId
+
+  /**
+   * Name of the chain as defined in
+   */
+  readonly name: string
+
+  /**
+   * Label of the chain. Field used for display purposes.
+   */
+  readonly label: string
+
+  /**
+   * Native currency of the chain.
+   */
+  readonly nativeCurrency: TokenInfo
+
+  /**
+   * ERC-3770 address prefix
+   *
+   * See https://eips.ethereum.org/EIPS/eip-3770
+   */
+  readonly addressPrefix: string
+
+  /**
+   * Whether the chain is a testnet.
+   */
+  readonly isTestnet: boolean
+
+  /**
+   * Contracts of the chain.
+   */
+  readonly contracts: ChainContracts
+
+  /**
+   * Main color of the chain, used for presentation purposes.
+   */
+  readonly color: string
+
+  /**
+   * Logo of the chain.
+   */
+  readonly logo: { light: string; dark: string }
+
+  /**
+   * Documentation of the chain.
+   */
+  readonly docs: WebUrl
+
+  /**
+   * Website of the chain.
+   */
+  readonly website: WebUrl
+
+  /**
+   * Block explorer of the chain.
+   */
+  readonly blockExplorer: WebUrl
+
+  /**
+   * Bridges of the chain.
+   */
+  readonly bridges?: WebUrl[]
+
+  /**
+   * RPC URLs of the chain.
+   */
+  readonly rpcUrls: {
+    [key: string]: ChainRpcUrls
+    default: ChainRpcUrls
+  }
+}
+
+const CHAIN_ASSETS = 'https://raw.githubusercontent.com/cowprotocol/cowswap/refs/heads/develop/libs/assets/src/cow-swap'
```

Comment:
TODO: Not use a reference from cowswap repo in the sdk repo.

Reply:
Yes, I had it in my notes, but adding a TODO